### PR TITLE
PERF: Do not call check_fields_exist inside get_documents.

### PR DESCRIPTION
**Pull request by danielballan**
The purpose of this line was to alert the user if none of the headers
had a field that the user requested. This of course requires greedily
checking all of the headers, which breaks the laziness of Results.
Based on benchmarking by @ordirules, it can take > 5 minutes to get
the first result out of month's worth of data. Removing this line
reduces that time to < 1 second.

### CODE REVIEW

**Comment by jrmlhermitte** [comment]
From Issue #299 
Thanks @danielballan ! :-)

**Comment by jrmlhermitte** [comment]
Here is comparison of 1 month's worth of data at cms:
```python
# old method
# commit d78bbfab6e21b6b7aa0eff61fec9ad24b7faac1e (currently 'master')
In [1]: import time

In [2]: hdrs = cmsdb(start_time="2017-06-02", stop_time="2017-07-01")

In [3]: stream = cmsdb.restream(hdrs)

In [4]: t1=time.time();nds = next(iter(stream));t2=time.time()
print(t2-t1)

In [5]: print(t2-t1)
453.14725065231323


# new method:
# commit : 81b4c3cccf5daa8edb33d1e48c93cd957c9a795c (currently Dan's branch)
In [1]: import time

In [2]: hdrs = cmsdb(start_time="2017-06-02", stop_time="2017-07-01")

In [3]: stream = cmsdb.restream(hdrs)

In [4]: t1=time.time();nds = next(iter(stream));t2=time.time()

In [5]: print(t2-t1)
0.2851831912994385
```
(didn't use `timeit`because results get cached after first run it seems). So that's 7min 33s or so versus .29 s...

EDIT : Timed 1 month worth of data instead for effect.

**Comment by jrmlhermitte** [comment]
EDIT : deleted this comment as it is out of scope (has to do with a bad document). Will post as a new issue soon

**Comment by jrmlhermitte** [comment]
The other issue I mentioned is resolved by #303 (still to be discussed).

I had noticed some other odd behaviour, but now I'm convinced this is fine. Basically, my search had a `start` document that had no `stop`. Before this PR, my `restream` would raise a `TypeError` upon the `restream` call, since the operation wasn't lazy. After this PR, `restream` only raises an error upon iterating on the bad document.

In other words, this looks good to me and works testing with my data. I don't think a test can be written for this one?
